src/orquestra/quantum/distributions/_measurement_outcome_distribution.py

```diff
@@ -76,6 +77,23 @@ def get_number_of_subsystems(self) -> int:
             list(self.distribution_dict.keys())[0]
         )  # already checked in __init__ that all keys have the same length
 
+    def subdistribution(counts, active_qubits):
```

Comment:
Given this is a class method, it should take `self` as a first argument and not use provided `counts` but rather `self.distribution_dict`. 
So sth like this (I added typing as well):
```suggestion
    def subdistribution(self, active_qubits: List[int]) -> MeasurementOutcomeDistribution:
```

Then at the end instead of return `new_counts` it should return `MeasurementOutcomeDistribution(new_counts)`.

Reply:
@Renaud2002  Your changes in https://github.com/zapatacomputing/orquestra-quantum/pull/29/commits/91fbbbe4c845a5a89428e37bb2148732bd30103e
 don't resolve this comment. You also need to refactor the body of the function t ouse `self.distribution_dict` and change the return statement.